app/state/reducers/reservationsRateReportModalReducer.js

```diff
@@ -0,0 +1,24 @@
+import { handleActions } from 'redux-actions';
+
+import uiActions from 'actions/uiActions';
+import actionTypes from 'api/actionTypes';
+
+const initialState = {
+  show: false,
+  units: [],
+};
+
+const hide = state => ({
+  ...state,
+  show: false,
+  units: [],
+});
+
+export default handleActions({
+  [uiActions.showReservationsRateReportModal]: state => ({
+    ...state,
+    show: true,
+  }),
+  [uiActions.hideReservationsRateReportModal]: hide,
+  [actionTypes.RESERVATIONS_RATE_REPORT]: () => {},
+}, initialState);
```

Comment:
Here you are mixing the use of functions and the settings state in the action handler. I would recommend to pick just one. So either create a `show` function and use that or then refactor the `hide` function here.
The `hide` function could actually be simplified here as well to just `state => initialState`.

Reply:
:+1: